### DISCUSSION TITLE
I am using Rust connector  try to deserialize Decimal type but not success

### DISCUSSION THREAD

**Discussion by measproem** [tarantool/tarantool#8653] [Q&A]
I am developing on top of the crate [Alopecosa crate](https://crates.io/crates/alopecosa). I have added a missing feature to support UUID, DateTime and Decimal types. Both UUID and DateTime work fine when I pack a message page to Tarantool and select it back. The problem is with Decimal type. I can pack and send it to Tarantool and insert it into a space that has a field of type decimal. But when I try to deserialize it, I get an error.

I debugged it and saw something like this:

```
localhost:3301> box.space.test:format()
---
- [{'name': 'amount', 'type': 'decimal'}]
...
localhost:3301> box.space.test:select()
---
- - [34904.075]
...
```

Then in my deserialize  Rust output (34904.075):

```
[2, 52, 144, 64, 124]
```

Based on my debugging, index 0 = 2 is the number of scale, but I don't know what the other numbers are. The length is not fixed; it depends on the field `amount`.

My problem is related to deserializing decimal only.

---------------------

Here my working code Deserialize  DateTIme just for info:
```

pub fn deserialize_datetime_tarantool<'de, D>(deserializer: D) -> Result<NaiveDateTime, D::Error>
where
    D: Deserializer<'de>,
{
    #[derive(Deserialize)]
    struct _ExtStruct((c_char, serde_bytes::ByteBuf));
    let _ExtStruct((_kind, bytes)) = serde::Deserialize::deserialize(deserializer)?;
    let data = bytes.into_vec();

    if data.len() != 8 && data.len() != 16 {
        return Err(serde::de::Error::invalid_length(data.len(), &"8 or 16 bytes"));
    }

    let seconds = i64::from_le_bytes([
        data[0], data[1], data[2], data[3], data[4], data[5], data[6], data[7],
    ]);
    let nanos = if data.len() == 8 {
        u32::from_le_bytes([0, 0, 0, 0])
    } else {
        u32::from_le_bytes([
            data[8], data[9], data[10], data[11],
        ])
    };

    let timestamp = NaiveDateTime::from_timestamp_opt(seconds, nanos).ok_or_else(|| serde::de::Error::custom("invalid datetime"))?;
    Ok(timestamp)
}
```

**Comment by Totktonada** [comment]
`[2, 52, 144, 64, 124]` doesn't look as `34904.075`:

```
tarantool> require('msgpack').encode(require('decimal').new('34904.075')):hex()
---
- c7060103034904075c
...
```

You don't need to guess the encoding, it is described in the documentation: https://www.tarantool.io/en/doc/latest/dev_guide/internals/msgpack_extensions/#the-decimal-type.

*(It seems, this question is connected with https://stackoverflow.com/q/76249086/1598057.)*

**Comment by measproem** [comment]
Thank I fixed with import `tarantool::decimal::Decimal;` the crate [Rust tarantool runtime ](https://crates.io/crates/tarantool) 
Just make Rust struct field as type `Decimal` from above crate it done
